keybase/platform_windows.go

```diff
@@ -62,6 +66,63 @@ func (c context) BeforeUpdatePrompt(update updater.Update, options updater.Updat
 	return nil
 }
 
+type regUninstallGetter func(string, Log) bool
+
+// It turns out a Wix bundle .exe supports "/layout", which among other things
+// generates a log of what it would do. We can parse this for Dokan product code variables,
+// which will be in the registry if the same version is already present.
+func CheckCanBeSilent(path string, log Log, regFunc regUninstallGetter) bool {
+	tempName, err := util.WriteTempFile("keybaseInstallLayout", []byte{}, 0700)
+	if err != nil {
+		log.Errorf("CheckCanBeSilent: Unable to write temp file")
+		return false
+	}
+	defer util.RemoveFileAtPath(tempName)
+
+	command := exec.Command(path, "/layout", "/quiet", "/log", tempName)
+	if err = command.Run(); err != nil {
+		log.Errorf("CheckCanBeSilent: Unable to execute %s", path)
+		return false
+	}
+
+	file, err := os.Open(tempName)
+	if err != nil {
+		log.Errorf("CheckCanBeSilent: Unable to open %s", tempName)
+		return false
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	codeFound := false
+	// Of the form: "Variable: DokanProduct64 = {65A3A964-3DC3-0100-0000-160621082245}"m
+	re := regexp.MustCompile(`Variable: DokanProduct(64|86) = (\{([[:xdigit:]]|-)+\})`)
+	for scanner.Scan() {
+		// Give ourselves a way to override silent install in the future
+		if strings.Contains(scanner.Text(), "Variable: KeybaseForceUI") {
+			log.Info("CheckCanBeSilent: KeybaseForceUI")
```

Comment:
This seems like Debug. And I always like to make my log messages read like normal writing.

E.g. log.Debug("Found KeybaseForceUI variable")